k4FWCore/include/k4FWCore/Consumer.h

```diff
@@ -90,6 +92,31 @@ namespace k4FWCore {
 
       // ... instead, they must implement the following operator
       virtual void operator()(const In&...) const = 0;
+
+      /**
+       * @brief    Get the input locations for a given input index
+       * @param i  The index of the input
+       * @return   A range of the input locations
+       */
+      const auto inputLocations(int i) const {
+        if (i >= sizeof...(In)) {
+          throw std::out_of_range("Called inputLocations with an index out of range");
+        }
+        return m_inputLocations[i] | std::views::transform([](const DataObjID& id) -> const auto& { return id.key(); });
+      }
+      /**
+       * @brief       Get the input locations for a given input name
+       * @param name  The name of the input
+       * @return      A range of the input locations
+       */
+      const auto inputLocations(std::string_view name) const {
+        auto it = std::find_if(m_inputLocations.begin(), m_inputLocations.end(),
```

Comment:
```suggestion
        auto it = std::ranges::find_if(m_inputLocations,
```
Since we can use c++20 now.

Reply:
Changed